packages/file_selector/file_selector_windows/lib/file_selector_windows.dart

```diff
@@ -3,16 +3,23 @@
 // found in the LICENSE file.
 
 import 'package:file_selector_platform_interface/file_selector_platform_interface.dart';
-
+import 'package:file_selector_windows/src/dart_file_selector_api.dart';
 import 'src/messages.g.dart';
 
 /// An implementation of [FileSelectorPlatform] for Windows.
 class FileSelectorWindows extends FileSelectorPlatform {
-  final FileSelectorApi _hostApi = FileSelectorApi();
+  /// Constructor for filePicker.
+  FileSelectorWindows([this._dartFileSelectorAPI]) {
+    _dartFileSelectorAPI = _dartFileSelectorAPI ?? DartFileSelectorAPI();
+    _api = _dartFileSelectorAPI!;
+  }
+
+  late DartFileSelectorAPI _api;
+  late DartFileSelectorAPI? _dartFileSelectorAPI;
```

Comment:
Could this be simplified to something like this?

```suggestion
  FileSelectorWindows(DartFileSelectorApi? api)
    : _api = api ?? DartFileSelectorApi();

  DartFileSelectorApi _api;
```

Reply:
Same comment for all other constructors in this pull request